src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -63,13 +106,60 @@ public void SubjectBuffersConnected(ITextView textView, ConnectionReason reason,
             }
 
             RazorLSPTextViewFilter.CreateAndRegister(vsTextView);
+
+            // Initialize the user's options and start listening for changes.
+            _razorEditorOptions = _editorOptionsFactory.GetOptions(textView);
+            Assumes.Present(_razorEditorOptions);
+            RazorOptions_OptionChanged(null, null);
+            _razorEditorOptions.OptionChanged += RazorOptions_OptionChanged;
         }
 
         public void SubjectBuffersDisconnected(ITextView textView, ConnectionReason reason, IReadOnlyCollection<ITextBuffer> subjectBuffers)
         {
             // When the TextView goes away so does the filter.  No need to do anything more.
         }
 
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        private async void RazorOptions_OptionChanged(object sender, EditorOptionChangedEventArgs e)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+        {
+            // Retrieve current space/tabs settings from from Tools->Options.
+            var (insertSpaces, tabSize) = GetRazorEditorOptions(_textManager);
+
+            // Update settings in the actual editor.
+            _razorEditorOptions.SetOptionValue(DefaultOptions.ConvertTabsToSpacesOptionId, insertSpaces);
```

Comment:
Hmm, ya this is interesting because with multiple text views this type will continually be reset